rewrite-java/src/main/java/org/openrewrite/java/cleanup/SimplifyConsecutiveAssignments.java

```diff
@@ -186,15 +186,19 @@ private String singleVariableName(Expression e) {
             private Statement combine(Statement s, String op, Expression right) {
                 if (s instanceof J.Assignment) {
                     J.Assignment assign = (J.Assignment) s;
+                    //noinspection ConstantConditions
                     J.Assignment after = s.withTemplate(combinedAssignment, s.getCoordinates().replace(),
-                            assign.getAssignment(), op, right);
+                            TypeUtils.asPrimitive(assign.getType()).getKeyword(), assign.getAssignment(), op, right);
```

Comment:
If the expression involves a boxed type like `Integer` rather than a primitive `int` this would be problematic. `TypeUtils.asPrimite()` would return `null`. I believe the "ConstantConditions" warning you're suppressing here is telling you about a legitimate problem

Reply:
yes, I agree, I reverted this change and fixed it by suppressing type validation since the left-hand side of the statement is ignored.